core/src/main/kotlin/com/justai/jaicf/generic/Compositions.kt

```diff
@@ -0,0 +1,42 @@
+package com.justai.jaicf.generic
+
+import com.justai.jaicf.api.BotRequest
+import com.justai.jaicf.context.ActivatorContext
+import com.justai.jaicf.reactions.Reactions
+
+// -- Type tokens' compositions
+
+private typealias Act<A> = ActivatorTypeToken<A>
+private typealias Ch<B, R> = ChannelTypeToken<B, R>
+private typealias Ctx<A, B, R> = ContextTypeToken<A, B, R>
+
+private typealias Req = BotRequest
+private typealias React = Reactions
+private typealias ActCtx = ActivatorContext
+
+// ---- ActivatorTypeToken and ActivatorTypeToken
+infix fun <A1: ActCtx, A2: A1> Act<A1>.and(other: Act<A2>): Act<A2> = other
```

Comment:
Why should we have such the "same type" tokens infix `and`? Maybe it could be better to signal the user that it's prohibited operator to prevent using this construction?

Reply:
No problems arise by using such overloads, and I don't think it must be prohibited, it looks like a limitation without an actual reason. They just exist for consistency and do nothing.

Reply:
Well, it looks confusing for me because it's not clear to resolve which one of two tokens will be used.

Reply:
The method's signature says that the most specific type will be used.
I can describe it in a comment before all compositions, as there's common logic for all methods: 
* the second argument always must be more (or no less) specific by all arguments than the first one
* the result is always a type-token of the most specific types